File: de thi HSG/2016-2017/K11/BAITHI/Ti01/subseq.cpp
#include <fstream>
using namespace std;
int main()
{
    long a[10001];
    long n, k, dem, dem_max;
    ifstream fi ("subseq.inp");
    ofstream fo ("subseq.out");
    
    fi>>n>>k;
    if (k==1) dem_max = n;
    else
    {
    	for (long i = 0; i<n; i++) fi>>a[i];
    	a[n] = k+1;
    
    	dem_max = 0;
    	dem = 0;
    	for (long i = 0; i<=n; i++)
    	if ((a[i] % k ) == 0) dem++;
   		else
    	{
    	if (dem > dem_max)
    	{
    		dem_max = dem;
    		dem = 0;
    	}
    	}
    }
    
    fo<<dem_max;
    
    fi.close();
    fo.close();
}

File: de thi HSG/2016-2017/K11/BAITHI/GK/subseq.pas
const MaxN = 10000;

var
    a: array[1..MaxN+1] of longint;
    n,k, dem_max: longint;


procedure nhap;
var i: longint;
begin
   readln(n,k);
   for i:=1 to n do read(a[i]);
   a[N+1] := k+1; // hang rao
end;
//-----------------------------------
procedure xuli;
var i,dem: longint;
begin
  dem_max := 0; dem := 0;
  for i:=1 to n+1 do
      if (a[i] mod k = 0) then inc(dem)
      else
          if dem > dem_max then begin dem_max := dem; dem := 0 end;

  write(dem_max);
end;
//-----------------------------------

begin
  assign(input,'subseq.inp'); reset(input);
  assign(output,'subseq.out'); rewrite(output);
  nhap;
  xuli;
  close(input); close(output);
end.
